EKF/control.cpp

```diff
@@ -976,10 +979,9 @@ void Ekf::controlHeightFusion()
 				_hgt_sensor_offset = _gps_sample_delayed.hgt - _gps_alt_ref + _state.pos(2);
 			}
 		}
-	}
+		break;
 
-	// set the height data source to range if requested
-	if ((_params.vdist_sensor_type == VDIST_SENSOR_RANGE) && _rng_hgt_valid) {
```

Comment:
The check for the `_rng_hgt_valid` is missing now

Reply:
Oops, nice catch. Fixed.

I also changed it to _range_data_ready && _rng_hgt_valid for consistency.